src/EditorFeatures/Core/Implementation/NavigationBar/NavigationBarController.cs

```diff
@@ -76,18 +73,12 @@ internal partial class NavigationBarController : ForegroundThreadAffinitizedObje
                     null));
 
             _selectedItemInfoTask = Task.FromResult(new NavigationBarSelectedTypeAndMember(null, null));
-
-            if (_workspaceRegistration.Workspace != null)
-            {
-                ConnectToWorkspace(_workspaceRegistration.Workspace);
```

Comment:
@jasonmalinowski rather than move the add/remove dropdown here, I moved the workspace registration listening up since the controller lives in the features layer

Reply:
Good idea!